controlplane/kubeadm/internal/workload_cluster_test.go

```diff
@@ -474,6 +474,85 @@ kubernetesVersion: v1.16.1
 	}
 }
 
+func TestUpdateImageRepositoryInKubeadmConfigMap(t *testing.T) {
+	kubeadmConfig := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      kubeadmConfigKey,
+			Namespace: metav1.NamespaceSystem,
+		},
+		Data: map[string]string{
+			clusterConfigurationKey: `
+apiVersion: kubeadm.k8s.io/v1beta2
+kind: ClusterConfiguration
+imageRepository: k8s.gcr.io
+`,
+		},
+	}
+
+	kubeadmConfigNoKey := kubeadmConfig.DeepCopy()
+	delete(kubeadmConfigNoKey.Data, clusterConfigurationKey)
+
+	kubeadmConfigBadData := kubeadmConfig.DeepCopy()
+	kubeadmConfigBadData.Data[clusterConfigurationKey] = `foobar`
+
+	g := NewWithT(t)
+	scheme := runtime.NewScheme()
+	g.Expect(corev1.AddToScheme(scheme)).To(Succeed())
+	tests := []struct {
+		name      string
+		imgRepo   string
+		objs      []runtime.Object
```

Comment:
objects

Reply:
do you want me to change all the references of `objs` to `objects` in this file.

Reply:
ok, let's leave it objs given the other tests use that.